### ISSUE TITLE
[FEATURE] Get listings using max_updated_at/min_updated_at filters in getListings API

### ISSUE THREAD

**Issue by subash047** [etsy/open-api#301]
**It is really frustrating that the Etsy API does not support getting listings(active, draft, inactive) by using max_updated_at or min_updated_at filter.**

Currently while getting listings for the Etsy store we need to get all the listings(active, draft, and inactive) for the initial date time. If we can filter the listings using max_updated_at or min_updated_at then we can save lots of API calls to the Etsy store. This will make it Etsy developer more feasible to perform regular sync between the Etsy store and other inventory management systems. 
It will be greatly helpful if Etsy API supports webhooks events for particular actions too. 

**For example, if listings get updated in Etsy then the updated event is triggered and the developer can check the listing updated event and perform necessary action to update listings on a separate inventory management system.**

For now to sync the Etsy listings we are performing an overall sync process that will take lots of time and also lots of API usage.

**Comment by Gareth064** [comment]
If webhooks aren't the top priority for new features then I will be shocked and appalled at the Etsy API team  😂

The following is assuming you are querying a shop as an authenticated user since you want to see other listings other than active....

As for your issue and the issue of making lots of API calls everytime you want to sync recently updated listings for your shop, I feel that until they add the query params for min/max_uodated etc, then a simple solution is to do the following.

Call getListingsByShop with the query param "state" to narrow it down to active, expired, draft etc. You might have to call this a couple of times to get all the listings via pagination.

Once you have all listings as a list in memory in your app, apply a filter against that list to reduce it to only listings which match the min/max_updated criteria, discarding the listings your not interested in.

Now process the updated listings as you require.

So from an API call perspective, you wil have a minimum of 1 API call to get all the listings, add another few more if you have to do paging. I'm going to assume your shop doesn't have 1000s of items so paging shouldn't be too much of an issue.

So in summary, getting all listings first should only be a couple of API calls, then just do the rest in memory/app as you'd normally would.

**Comment by etsyachristensen** [comment]
@subash047 This is a good suggestion. At the moment we can't really fulfill this wish but I can say that you could hit the `getListingsByShop` endpoint [here](https://developers.etsy.com/documentation/reference#operation/getListingsByShop) and pass `sort_on` as 'updated' and `sort_order` as 'desc'. This will sort all the listings in order of most recently updated. if you have a way to store this data on your end, then you can simply compare the `last_modified_timestamp` on each listing to what you have for that record in your system. If the dates don't match you can assume a change has been made. That endpoint also accepts `limit` and `offset` variables so you can page through it if you need to. It may not be perfect, but hopefully this helps.

**Comment by aziret** [comment]
Hey @etsyachristensen!

Thanks for your suggestion to use `sort_on = updated` and `sort_order = desc` params to get updated listings first.

I have a question regarding this approach: How do you distinguish listings if multiple of them have the same `last_modified_timestamp`.
Image a case when there are 200 listings with the same dates in `last_modified_timestamp` and those timestamps are not updated between the requests. Let's say I requested first page of the listings `limit = 100`. Is there any guarantee that I won't get the same (or some) listings from the first page on the second page when I use `limit = 100` AND `offset = 100`.

Is it even possible to have multiple listings with the same `last_modified_timestamp`?

Many thanks!

**Comment by etsyachristensen** [comment]
For a time there, yes you could have gotten the same listing on multiple pages. But we made a change recently to ensure that the sorting goes by the field you are requesting first and then we sort by id after so you are sure to never get duplicates in the different pages.

Yes it's *possible* to have multiple listings with the same last modified timestamp but highly unlikely.

**Comment by aziret** [comment]
That's great! Thank you!
